scripts/west_commands/sdk.py

```diff
@@ -0,0 +1,368 @@
+# Copyright (c) 2024 TOKITA Hiroshi
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import hashlib
+import os
+import platform
+from py7zr import SevenZipFile
+import re
+import requests
+import subprocess
+import tarfile
+import tempfile
+import textwrap
+
+from west.commands import WestCommand
+
+
+class Sdk(WestCommand):
+    def __init__(self):
+        super().__init__(
+            "sdk",
+            "manage Zephyr SDK",
+            "List and Install Zephyr SDK",
+        )
+
+    def do_add_parser(self, parser_adder):
+        parser = parser_adder.add_parser(
+            self.name,
+            help=self.help,
+            description=self.description,
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            epilog=textwrap.dedent(
+                """
+            Listing SDKs:
+
+                Run 'west sdk' command information about available SDKs is displayed.
+
+
+            Installing SDK:
+
+                Run 'west sdk --install=latest' to install the latest SDK.
+
+                Set --install option to install the specified version of the SDK.
+                The 'latest' can be used instead of the version tag. In this case,
+                install the latest version of Zephyr-SDK.
+                If it is already installed, it will be used to continue processing.
+
+                You can specify the installation directory with --install-base.
+                The SDK will be extracted under the specified directory.
+
+                The --toolchains and --hosttools options specify the installation modules.
+                If neither is specified, an interactive installer will be launched.
+
+                --api-url specifies the REST API endpoint for GitHub releases information
+                when installing the SDK from a different GitHub repository.
+
+            Note:
+
+                To extract 7z format files for Windows version SDK, you need the '7z' command.
+                You can get it from https://7-zip.org/download.html."""
+            ),
+        )
+
+        parser.add_argument(
+            "--install",
+            default=None,
+            dest="sdk_version",
+            metavar="SDK_VER",
+            help="Install the selected version of Zephyr SDK. "
+            'If you set this option to "latest", select the latest version.',
```

Comment:
there's a `SDK_VERSION` file at the top of the main repository, that is the one used by CI so it's in a way the "current" one for the current checkout, how about a way of picking that version? once it's deployed it simplify https://github.com/zephyrproject-rtos/action-zephyr-setup/blob/main/action.yml a lot

Reply:
I changed it to refer to the SDK_VERSION by default.
Also, Add `--install-dir` option for considering the CI use case.
This option allows the modification of the SDK install destination.